airbyte-webapp/src/components/connectorBuilder/Builder/PaginationSection.tsx

```diff
@@ -0,0 +1,180 @@
+    >
+      <BuilderField
+        type="enum"
+        path={streamFieldPath("paginator.pageTokenOption.inject_into")}
+        options={injectIntoValues}
+        label="Inject into"
+        tooltip="Configures where the page token should be set on the HTTP requests"
+      />
+      <BuilderField
+        type="string"
+        path={streamFieldPath("paginator.pageTokenOption.field_name")}
+        label="Field name"
+        tooltip="Configures which key should be used in the location that the page token is being injected into"
+        optional
+      />
+    </GroupControls>
+  );
+
+  const pageSizeOption = (
+    <GroupControls
+      label={
+        <ControlLabels
+          label="Page size option"
+          infoTooltipContent="Configures how the page size will be sent in requests to the source API"
+        />
+      }
+    >
+      <BuilderField
+        type="enum"
+        path={streamFieldPath("paginator.pageSizeOption.inject_into")}
+        options={injectIntoValues}
+        label="Inject into"
+        tooltip="Configures where the page size should be set on the HTTP requests"
+      />
+      <BuilderField
+        type="string"
+        path={streamFieldPath("paginator.pageSizeOption.field_name")}
+        label="Field name"
+        tooltip="Configures which key should be used in the location that the page size is being injected into"
+        optional
+      />
+    </GroupControls>
+  );
+
+  return (
+    <BuilderCard
+      toggleConfig={{
+        label: (
+          <ControlLabels
+            label="Pagination"
+            infoTooltipContent="Configure how pagination is handled by your connector"
+          />
+        ),
+        toggledOn,
+        onToggle: handleToggle,
+      }}
+    >
+      <BuilderOneOf
+        path={streamFieldPath("paginator.strategy")}
+        label="Mode"
+        tooltip="Pagination method to use for requests sent to the API"
+        options={[
+          {
+            label: "Offset Increment",
+            typeValue: "OffsetIncrement",
+            children: (
+              <>
+                <BuilderField
+                  type="number"
+                  path={streamFieldPath("paginator.strategy.page_size")}
+                  label="Page size"
+                  tooltip="Size of pages to request from API"
+                />
+                {pageTokenOption}
+                {pageSizeOption}
+              </>
+            ),
+          },
+          {
+            label: "Page Increment",
+            typeValue: "PageIncrement",
+            children: (
+              <>
+                <BuilderField
+                  type="number"
+                  path={streamFieldPath("paginator.strategy.page_size")}
+                  label="Page size"
+                  tooltip="Size of pages to request from API"
+                />
+                <BuilderField
+                  type="number"
+                  path={streamFieldPath("paginator.strategy.start_from_page")}
```

Comment:
Side note: I just realized that for these it's very common to interpolate a value from the state, but it's not possible right now as this is a number input. 

Seems like https://github.com/airbytehq/airbyte/issues/20581 is not just a UX improvement but necessary for adding certain functionality in cases like this.